youtube_dl/YoutubeDL.py

```diff
@@ -948,7 +953,14 @@ def iter_playlistitems(format):
                 entry_result = self.process_ie_result(entry,
                                                       download=download,
                                                       extra_info=extra)
+
+                entry_result_uploaddate = entry_result.get('upload_date')
+                if entry_result_uploaddate:
+                    if self.params.get('date_playlist_order') in ('desc', 'asc') and entry_result_uploaddate not in self.params.get('daterange'):
```

Comment:
No, it's not the point. It **must not** skip items than belong to the range specified.